File: ajax-java/src/main/java/fr/isima/cours/jee/controllers/HeureMVCController.java
package fr.isima.cours.jee.controllers;

import fr.isima.cours.jee.service.HourGenerator;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.servlet.ModelAndView;

@Controller
@RequestMapping("/heure")
public class HeureMVCController {

    private final HourGenerator generator;

    @Autowired
    public HeureMVCController(HourGenerator generateurHeureCourante) {
        this.generator = generateurHeureCourante;
    }

    @GetMapping
    public ModelAndView heure() {
        return new ModelAndView("heureCourante", "currentDateAsString", this.generator.nowWithStandardFormat().getFormatted());
    }

}

File: ajax-kotlin/src/main/kotlin/fr/isima/controllers/HeureMVCController.kt
package fr.isima.controllers

import fr.isima.service.HourGenerator
import fr.isima.view.toHtml
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseBody

@Controller
@RequestMapping("/heure")
class HeureMVCController @Autowired
constructor(private val generator: HourGenerator) {

    @GetMapping
    @ResponseBody
    fun heure() = this.generator.nowWithStandardFormat().toHtml()

}
